### ISSUE TITLE
Use custom ObjectMapper

### ISSUE THREAD

**Issue by xerial** [komamitsu/fluency#30]
For using fluency in Scala, I need to have additional modules in ObjectMapper. For example, 
https://github.com/FasterXML/jackson-module-scala
This module is useful for generating json (msgpack) log from Scala case classes.

In the current code, ObjectMapper is explicitly instantiated here:
https://github.com/komamitsu/fluency/blob/b87431ec7f4d1370fb1fe84d5f4e61ef248e95ae/src/main/java/org/komamitsu/fluency/buffer/Buffer.java#L28

So we have no way to configure ObjectMapper modules. 

@komamitsu 
Do you have any idea on making it extensible? I think some Provider of ObjectMapper is necessary (as in Guice: https://github.com/google/guice/wiki/ProviderBindings)

And also using thread local storage as an ObjectMapper holder is not always a good idea, especially if we have hundreds of threads. Using some session-scoped singleton would be better.

**Comment by komamitsu** [comment]
I see. It sounds useful to be able to pass a customized ObjectMapper. I'll try to address it.

RE thread local, you have a point. BTW, what did you mean by `session-scoped singleton`? Could you give me pointers to it?

**Comment by komamitsu** [comment]
I just tried removing the thread local cache. It got 5-6 times slower without the cache...

**Comment by xerial** [comment]
Caching is must have, but I mean caching to thread local storage is not good since it creates many ObjectMapper instances although ObjectMapper is immutable and safe to share.

**Comment by komamitsu** [comment]
Added org.komamitsu.fluency.buffer.Buffer.Config#setJacksonModules in https://github.com/komamitsu/fluency/pull/32. You can register Jackson modules in ObjectMapper with it.

**Comment by komamitsu** [comment]
@xerial I released fluency 1.0.0 that includes a new API `PackedForwardBuffer.Config#setJacksonModules` which makes it able to pass Jackson modules like this
```
new Fluency.Builder(new TCPSender.Config().createInstance()).
    setBufferConfig(
        new PackedForwardBuffer.Config().
            setJacksonModules(Arrays.asList(DefaultScalaModule))).build();
```
Can I ask you to try it?

**Comment by xerial** [comment]
Sure. I'll do
